### ISSUE TITLE
Error in lowering mrcnn model 

### ISSUE THREAD

**Issue by ArunaKote** [iree-org/iree#6893]
Added model signature to model using below code:

```python
import tensorflow as tf
import tensorflow_hub as hub

model_url = "https://tfhub.dev/tensorflow/mask_rcnn/inception_resnet_v2_1024x1024/1"
loaded_model = hub.load(model_url)
call = loaded_model.__call__.get_concrete_function(
tf.TensorSpec(shape=(1, 1024, 1024, 3), dtype=tf.uint8))
signatures = {'predict': call}
tf.saved_model.save(loaded_model,'/data/aruna/maskedrcnn/iree', signatures=signatures)
```

Then used /data/aruna/iree/integrations/tensorflow/bazel-bin/iree_tf_compiler/iree-import-tf -tf-import-type=savedmodel_v1 -tf-savedmodel-exported-names=predict /data/aruna/maskedrcnn/iree --print-ir-before-all&>2 -o sample.mlir

<unknown>:0: error: loc(callsite(callsite(callsite("BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/Select@__inference___call___42606" at "StatefulPartitionedCall@__inference_restored_function_body_108980") at "StatefulPartitionedCall@__inference_signature_wrapper_110151") at "StatefulPartitionedCall")): currently unsupported operand types: 'tensor<?xf32>' and 'tensor<300xf32>'
<unknown>:0: note: loc("StatefulPartitionedCall"): called from
Running iree-import-tf TF import pass pipeline failed

**Comment by dinkdeep** [comment]
I am looking at these one by one but looks like tf.ResizeBilinear/tf.NonMaxSuppressionV5 and tf.CropAndResize could be major ones, 

Actually started with tf.CropAndResize 
1) created a python equivalent code to use the python API ( tf.image.crop_and_resize) tf.image.crop_and_resize  |  TensorFlow Core v2.6.0   to get tf dilaect operator 
2) Created a saved_model.pb
 3) Ran the saved model with tflite_convert . 
4 Dumped the tf and tfl dialects using module.dump() and extracted the both these dialects on a txt file 
tfllie_convert failed with error converting tf.CropAndResize to equivalent tfl /data/Work/tf/tensorflow

time TF_DUMP_GRAPH_PREFIX=crop TF_CPP_MAX_VLOG_LEVEL=4 bazel-bin/tensorflow/lite/python/tflite_convert --saved_model_dir=/data/Work/dkd/tmp42f5h1vo/module_with_signature/ --output_file=crop.tflite &>log.crop

  tensorflow/compiler/mlir/lite/flatbuffer_export.cc:406] GetTensorFlowNodeDef %0 = "tf.CropAndResize"(%arg0, %arg1, %arg2, %arg3) {T = f32, device = "", extrapolation_value = 0.000000e+00 : f32, method = "bilinear"} : (tensor<?x?x?x?xf32>, tensor<?x4xf32>, tensor<?xi32>, tensor<2xi32>) -> tensor<?x10x10x?xf32> loc(callsite(callsite("CropAndResize@inference_call_43" at "PartitionedCall@inference_signature_wrapper_50") at "PartitionedCall")): error: 'tf.CropAndResize' op is neither a custom op nor a flex op 
2021-08-30 12:42:35.450948: I tensorflow/compiler/mlir/lite/flatbuffer_export.cc:1883] resourceops 0 error: failed while converting: 'main': Some ops are not supported by the native TFLite runtime, you can enable TF kernels fallback using TF Select. See instructions: Select TensorFlow operators  |  TensorFlow Lite 
 TF Select ops: CropAndResize, Placeholder Details: tf.CropAndResize(tensor<?x?x?x?xf32>, tensor<?x4xf32>, tensor<?xi32>, tensor<2xi32>) -> (tensor<?x10x10x?xf32>) : {T = f32, device = "", extrapolation_value = 0.000000e+00 : f32, method = "bilinear"} tf.Placeholder() -> (tensor<?x?x4xf32>) : {device = "", shape = #tf_type.shape<?x?x4>} tf.Placeholder() -> (tensor<?x?x?x?xf32>) : {device = "", shape = #tf_type.shape<?x?x?x?>}

tf-opt too failed while legalizing
 /data/dkd/tf_linalg/tensorflow/bazel-bin/tensorflow/compiler/mlir/tf-opt -tf-executor-island-coarsening -canonicalize --tf-device-decompose-resource-ops-in-cluster --tf-promote-var-handles-to-args --tf-readonly-references-to-resources --tf-resource-device-inference --tf-executor-to-functional-conversion --tf-shape-inference -xla-legalize-tf crop_tf.mlir -print-ir-before-all --mlir-disable-threading -print-ir-after-all &>log.cropResize crop_

tf.mlir:11:3: error: The following operations cannot be legalized: tf.CropAndResize (count: 1); tf.Placeholder (count: 2). These legalization failure(s) may be due to missing TF to HLO lowerings and/or unsupported attributes, etc. builtin.func private @inference_call_430(%arg0: tensor<2x200x200x3xui32> {tf._user_specified_name = "x"}) -> tensor<*xf32> attributes {tf._construction_context = "kEagerRuntime", tf._input_shapes = [#tf_type.shape<2x200x200x3>]} {\ ^ crop_tf.mlir:11:3: error: Emitting more detail about one op that failed to legalize... builtin.func private @inferencecall430(%arg0: tensor<2x200x200x3xui32> {tf._user_specified_name = "x"}) -> tensor<xf32> attributes {tf._construction_context = "kEagerRuntime", tf._input_shapes = [#tf_type.shape<2x200x200x3>]} { ^ crop_tf.mlir:38:11: error: 'tf.CropAndResize' op is not legalizable %12 = "tf.CropAndResize"(%0, %4, %11, %cst) {T = f32, device = "", extrapolation_value = 0.000000e+00 : f32, method = "bilinear"} : (tensor<?x?x?x?xf32>, tensor<xf32>, tensor<xi32>, tensor<2xi32>) -> tensor<xf32> 
crop_tf.mlir:38:11: note: see current operation: %71 = "tf.CropAndResize"(%9, %18, %70, %8) {T = f32, device = "", extrapolation_value = 0.000000e+00 : f32, method = "bilinear"} : (tensor<?x?x?x?xf32>, tensor<?x4xf32>, tensor<?xi32>, tensor<2xi32>) -> tensor<?x10x10x?xf32> // -----// IR Dump After LegalizeTF Failed //----- /(edited)

  Will try with iree-import-tf now.

**Comment by hanhanW** [comment]
It's a bit hard to parse contexts. Could you organize the error log better with a link or a [code block](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks)?

If there are two models, I think it's better to file a bug for each one. It's easier to us to track the missing work. :)

> Can you check if we can run it in TFLite?
Sorry that I did not point it out directly. I'm not familiar with TFLite MLIR path. I meant to run TFLite converter and inference (like benchmark_tool?) on the models.

> I also see these 6 tf Ops after the ConvertToMHLOPass which might need the equivalent HLO conversions proably

You can try to add some patterns to https://github.com/tensorflow/tensorflow/blob/800d21c92441c17a9ae521adcff62d2ce0640914/tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc

I remembered that we have TopKV2 lowering, but we could miss some cases.

**Comment by dinkdeep** [comment]
We will create separate issues and attach the log files and paste only the relevant error messages in the console .

Regarding TFLite converter  we did run the model with it and found that it was not able to lower the tf.CropAndResize(tensor<1x64x64x1088xf32>, tensor<?x4xf32>, tensor<100xi32>, tensor<2xi32>)  to an equivalent 
tfl Operator , We will reattach the error log for it . 

Please suggest the name of the benchmark_tool we can include it too.  

Sure we will try to see if we can add some patterns for the ops,  in case @benvanik  might sign up for tf.NonMaxSuppressionV5 as discussed over iree discord  .

**Comment by dinkdeep** [comment]
@ArunaKote Please attach the error log files and create separate issues .

**Comment by ArunaKote** [comment]
Ok
